### PR TITLE
Release/3.0.0

### PR DESCRIPTION
**Pull request by CassioMG**
## Summary

- Upgrade `@stellar/stellar-sdk` from `14.5.0` to `15.0.1` across all 3 packages — **Protocol 26 support**
- Upgrade `@stellar/freighter-api` from `^2.0.0` to `^6.0.1` in the key manager package
- Fix longstanding bug in freighter handler that hardcoded `Networks.PUBLIC` for XDR deserialization regardless of actual network
- Rewrite freighter handler for freighter-api v6 API changes (new `signTransaction` signature and response shape)
- Add freighter handler test coverage (previously zero tests)
- Add XDR integer boundary tests in the Soroban package for Protocol 26 strict validation
- Fix paging ID test that was failing due to test anchor server now rejecting invalid IDs

### All packages
- `@stellar/stellar-sdk` upgraded from `14.5.0` to `15.0.1` (Protocol 26 XDR)
- XDR integer construction now throws on overflow/underflow (upstream)
- `TransactionBase.networkPassphrase` setter now throws to enforce immutability (upstream)

### Key Manager (`@stellar/typescript-wallet-sdk-km`)
- `@stellar/freighter-api` upgraded from `^2.0.0` to `^6.0.1`
- Freighter handler: `custom.network` param replaced with `custom.networkPassphrase`

## Test plan

- [x] `yarn build` — all 3 packages compile successfully
- [x] `yarn lint` — zero warnings
- [x] `yarn test:ci` — 244 passed, 4 skipped, 0 failed
- [x] Freighter handler tests: 9 tests covering sign, errors, network passphrase resolution
- [x] Soroban XDR boundary tests: 10 tests covering i32/u32/i64/u64 min/max/overflow

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: @stellar/typescript-wallet-sdk-km/src/Handlers/freighter.ts [hunk 1]
```diff
@@ -24,23 +24,35 @@ export const freighterHandler: KeyTypeHandler = {
...
 
       // fromXDR() returns type "Transaction | FeeBumpTransaction" and
       // signTransaction() doesn't like "| FeeBumpTransaction" type, so casting
       // to "Transaction" type.
       return TransactionBuilder.fromXDR(
-        response,
-        Networks.PUBLIC,
+        response.signedTxXdr,
+        networkPassphrase,
       ) as Transaction;
```
[the remark this answers is not in the document]
Comment (CassioMG) [reply @stellar/typescript-wallet-sdk-km/src/Handlers/freighter.ts:49]: If Freighter returns { signedTxXdr: undefined } with no error, fromXDR will throw something like "Cannot read XDR from undefined". Adding a guard is defensive programming against an API contract violation. Not wrong to add, but it's a "belt and suspenders" situation — the Freighter API contract says signedTxXdr is always present when error is absent. The fromXDR error + our catch block will produce a reasonable error message anyway.

File: @stellar/typescript-wallet-sdk-km/src/Handlers/freighter.ts [hunk 2]
```diff
@@ -24,23 +24,35 @@ export const freighterHandler: KeyTypeHandler = {
       );
```
[the remark this answers is not in the document]
Comment (CassioMG) [reply @stellar/typescript-wallet-sdk-km/src/Handlers/freighter.ts:22]: fixed

File: @stellar/typescript-wallet-sdk-km/src/Handlers/freighter.ts [hunk 3]
```diff
@@ -18,29 +18,42 @@ export const freighterHandler: KeyTypeHandler = {
...
+
     try {
-      const response = await freighterApi.signTransaction(
-        transaction.toXDR(),
-        custom && custom.network ? custom.network : undefined,
-      );
+      const response = await freighterApi.signTransaction(transaction.toXDR(), {
+        networkPassphrase,
+        address: custom && custom.address ? custom.address : undefined,
+      });
```
[the remark this answers is not in the document]
Comment (CassioMG) [reply @stellar/typescript-wallet-sdk-km/src/Handlers/freighter.ts:32]: Interesting suggestion but wrong for this handler. The transaction param is a Transaction object that the SDK built — it already has a networkPassphrase from when it was constructed. But the Freighter handler's job is to send the XDR to the Freighter browser extension for signing. The key.network represents what network the user's Freighter wallet is configured for. These could legitimately differ (e.g., the SDK builds a testnet transaction, the key is configured for testnet). Using transaction.networkPassphrase as fallback is reasonable, but it's a behavioral change beyond what we scoped. The current fallback chain (custom.networkPassphrase > key.network > Networks.PUBLIC) was a deliberate design choice.

File: @stellar/typescript-wallet-sdk-km/src/Handlers/freighter.ts [hunk 4]
```diff
@@ -18,29 +18,42 @@ export const freighterHandler: KeyTypeHandler = {
...
-        Networks.PUBLIC,
+        response.signedTxXdr,
+        networkPassphrase,
       ) as Transaction;
     } catch (error) {
-      const errorMsg = error.toString();
+      if (error instanceof Error) {
+        throw error;
+      }
+      const errorMsg = String(error);
```
[the remark this answers is not in the document]
Comment (CassioMG) [reply @stellar/typescript-wallet-sdk-km/src/Handlers/freighter.ts:54]: intended

File: @stellar/typescript-wallet-sdk-km/CHANGELOG.MD [hunk 5]
```diff
@@ -1,3 +1,16 @@
+# Release notes - Typescript Wallet SDK Key Manager - 3.0.0
+
+### BREAKING CHANGES
+* Upgraded `@stellar/stellar-sdk` from `14.5.0` to `15.0.1` — Protocol 26 support
+* Upgraded `@stellar/freighter-api` from `^2.0.0` to `^6.0.1`
+* Freighter handler: `custom.network` replaced with `custom.networkPassphrase`
```
[the remark this answers is not in the document]
Comment (CassioMG) [reply @stellar/typescript-wallet-sdk-km/CHANGELOG.MD:6]: fixed

File: @stellar/typescript-wallet-sdk-soroban/CHANGELOG.MD [hunk 6]
```diff
@@ -1,3 +1,12 @@
+# Release notes - Typescript Wallet SDK Soroban - 3.0.0
+
+### BREAKING CHANGES
+* Upgraded `@stellar/stellar-sdk` from `14.5.0` to `15.0.1` — Protocol 26 support
+* XDR integer strict validation: i64/u64 `fromString()` throws on overflow/underflow, i32/u32 throws at XDR serialization (upstream change)
```
[the remark this answers is not in the document]
Comment (CassioMG) [reply @stellar/typescript-wallet-sdk-soroban/CHANGELOG.MD:6]: fixed

File: @stellar/typescript-wallet-sdk-km/package.json [hunk 7]
```diff
@@ -37,8 +37,8 @@
     "@ledgerhq/hw-transport-u2f": "^5.36.0-deprecated",
     "@stablelib/base64": "^2.0.0",
     "@stablelib/utf8": "^2.0.0",
-    "@stellar/freighter-api": "^2.0.0",
-    "@stellar/stellar-sdk": "14.5.0",
+    "@stellar/freighter-api": "^6.0.1",
```
Comment (piyalbasu) [inline @stellar/typescript-wallet-sdk-km/package.json]: Very NIT: we should pin the freighter-api dep to 6.0.1 rather than the floating version

**Review by piyalbasu — APPROVED** [review]
This PR is lgtm and the Claude code review came back good. I just left 1 very nit comment

File: @stellar/typescript-wallet-sdk-km/package.json (hunk 7 above)
Comment (CassioMG) [reply @stellar/typescript-wallet-sdk-km/package.json]: nice catch, I'll fix it

**Comment by CassioMG** [comment]
FYI, I had to re-create the commit history on this PR because the original commits were missing a `verified` signature but the PR stays exactly the same
